File: leetcode/easy/242-validAnagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        
        int[] charCount = new int[26];
        
        for (int i = 0; i < s.length(); i++) {
            charCount[(int) s.charAt(i) - 97]++;
            charCount[(int) t.charAt(i) - 97]--;
        }
        
        for (int n : charCount) {
            if (n != 0) {
                return false;
            }
        }
        return true;
    }
}

File: leetcode/easy/242-validAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) return false;

        int count[26] {0};

        for (int i = 0; i < s.length(); i++) {
            int index1 = (int) s[i] - 97;
            int index2 = (int) t[i] - 97;

            count[index1]++;
            count[index2]--;
        }

        for (const auto num: count) {
            if (num != 0) return false;
        }

        return true;
    }
};
